src/client/boundaries.rs

```diff
@@ -17,51 +18,76 @@ pub mod boundary_type {
     pub const CROSS: &str = "┼";
 }
 
+pub mod colors {
+    use ansi_term::Colour::{self, Fixed};
+    pub const WHITE: Colour = Fixed(255);
+    pub const GREEN: Colour = Fixed(154);
+    pub const GRAY: Colour = Fixed(238);
+}
+
 pub type BoundaryType = &'static str; // easy way to refer to boundary_type above
 
 #[derive(Clone, Copy, Debug)]
 pub struct BoundarySymbol {
     boundary_type: BoundaryType,
     invisible: bool,
+    color: Option<Colour>,
 }
 
 impl BoundarySymbol {
     pub fn new(boundary_type: BoundaryType) -> Self {
         BoundarySymbol {
             boundary_type,
             invisible: false,
+            color: Some(colors::GRAY),
         }
     }
     pub fn invisible(mut self) -> Self {
         self.invisible = true;
         self
     }
+    pub fn color(&mut self, color: Option<Colour>) -> Self {
+        self.color = color;
+        *self
+    }
 }
 
 impl Display for BoundarySymbol {
     fn fmt(&self, f: &mut Formatter) -> Result<(), Error> {
-        if self.invisible {
-            write!(f, " ")
-        } else {
-            write!(f, "{}", self.boundary_type)
+        match self.invisible {
+            true => write!(f, " "),
+            false => match self.color {
+                Some(_) => write!(f, "{}", self.color.unwrap().paint(self.boundary_type)),
```

Comment:
`Some(color) => write!(f, "{}", color.paint(self.boundary_type))` won't work?

Reply:
I think it should, don't know honestly why I left it this way, some things can't be explained 😄